server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -87,6 +89,60 @@ func (s *BuildBuddyServer) SearchInvocation(ctx context.Context, req *inpb.Searc
 	return searcher.QueryInvocations(ctx, req)
 }
 
+func (s *BuildBuddyServer) authorizeInvocationUpdate(ctx context.Context, reqCtx *ctxpb.RequestContext, in *tables.Invocation) error {
+	auth := s.env.GetAuthenticator()
+	if auth == nil {
+		return status.UnimplementedError("Not Implemented")
+	}
+	if _, err := auth.AuthenticatedUser(ctx); err != nil {
+		return err
+	}
+
+	// Invocations should probably never be writeable by OTHERS,
+	// but this check is included for completeness.
+	if in.Perms&perms.OTHERS_WRITE != 0 {
+		return nil
```

Comment:
as written, this silently fails -- do you want to log something when this happens?

Maybe

log.Print("Ignoring request to allow OTHERS_WRITE on invocation. This should not happen!")